src/components/Comment/index.spec.tsx

```diff
@@ -1,82 +1,309 @@
 import * as React from 'react';
-import { shallow } from 'enzyme';
+import { Button } from 'react-bootstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Store } from 'redux';
 
-import { createFakeComment, createFakeEvent } from '../../test-helpers';
+import configureStore from '../../configureStore';
+import { actions } from '../../reducers/comments';
+import {
+  createFakeExternalComment,
+  createFakeEvent,
+  createFakeThunk,
+  shallowUntilTarget,
+  spyOn,
+} from '../../test-helpers';
 import styles from './styles.module.scss';
 
-import Comment, { PublicProps } from '.';
+import Comment, { CommentBase, DefaultProps, PublicProps } from '.';
 
 describe(__filename, () => {
-  const render = (moreProps: Partial<PublicProps> = {}) => {
+  type RenderParams = { store?: Store } & Partial<PublicProps> &
+    Partial<DefaultProps>;
+
+  const render = ({
+    store = configureStore(),
+    commentId = null,
+    ...moreProps
+  }: RenderParams = {}) => {
     const props = {
-      comment: null,
-      readOnly: true,
+      addonId: 1,
+      commentId,
+      fileName: null,
+      line: null,
+      readOnly: false,
+      versionId: 2,
       ...moreProps,
     };
-    return shallow(<Comment {...props} />);
+    return shallowUntilTarget(<Comment {...props} />, CommentBase, {
+      shallowOptions: { context: { store } },
+    });
   };
 
-  it('renders a form when there is no comment', () => {
-    const root = render({ comment: null });
+  const dispatchComment = ({
+    store = configureStore(),
```

Comment:
Yeah, it could be useful for future test cases but, more importantly, adding this flexibility is not harder than doing without it. In other words, these two approaches would require the same lines of code:

```ts
  const dispatchComment = ({
    store = configureStore(),
    comment = createFakeExternalComment(),
    fileName = null,
    line = null,
    versionId = 1,
  } = {}) => {
    store.dispatch(actions.setComment({ fileName, line, versionId, comment }));
    return { store };
  };
```

```ts
  const dispatchComment = ({
    comment = createFakeExternalComment(),
    fileName = null,
    line = null,
    versionId = 1,
  } = {}) => {
    const store = configureStore();
    store.dispatch(actions.setComment({ fileName, line, versionId, comment }));
    return { store };
  };
```

...only one would be less flexible.